### ISSUE TITLE
issue with token validator

### ISSUE THREAD

**Issue by marius-xentral** [auth0/go-jwt-middleware#206]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/go-jwt-middleware#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://pkg.go.dev/github.com/auth0/go-jwt-middleware/v2) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/go-jwt-middleware/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/go-jwt-middleware/releases/latest) of this SDK and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Hi!
I am getting this error here from the token validator: 

`failed to deserialize token claims: could not get token claims: square/go-jose: unsupported key type/format`. 

The only thing I changed recently was upgrading go from 1.19 to 1.20, but running this in a 1.19 container also yields the same result by now. 

I used https://jwt.io/ to look at the token and the payload looks fine:
`{
  "https://xentral.com/license-id": "xxxx-xxxx-xxxx-xxxx",
  "iss": "https://xxx.eu.auth0.com/",
  ...
}`

I am basically using the default code from the docs:

```
type CustomClaims struct {
	LicenseId string `json:"https://xentral.com/license-id"`
}

func NewTokenValidator(authPlatformDomain string, audience string) (TokenValidator, error) {
	issuerURL, err := url.Parse(authPlatformDomain)
	if err != nil {
		return nil, err
	}
	provider := jwks.NewCachingProvider(issuerURL, 5*time.Minute)

	// We want this struct to be filled in with
	// our custom claims from the token.
	customClaims := func() validator.CustomClaims {
		return &CustomClaims{}
	}
	// Set up the validator.
	jwtValidator, err := validator.New(
		provider.KeyFunc,
		validator.RS256,
		issuerURL.String(),
		[]string{audience},
		validator.WithCustomClaims(customClaims),
	)
	if err != nil {
		return nil, err
	}

	return jwtValidator, nil
}
```
I also tried this with the current master branch since I saw that you got rid of the deprecated square/go-jose dependency, but I basically get the same error message. Feels like I am missing something very basic here ... 

### Reproduction

1. use the above code and custom claim
2. generate token
3. try to validate it

### Go JWT Middleware version

2.1.0

### Go version

1.20

**Comment by marius-xentral** [comment]
Here's also a list of the things that I am using in this project. maybe a dependency issue?
```
module analytics-service

go 1.20

require (
	github.com/DATA-DOG/go-sqlmock v1.5.0
	github.com/auth0/go-jwt-middleware/v2 v2.1.0
	github.com/aws/aws-sdk-go v1.44.269
	github.com/confluentinc/confluent-kafka-go/v2 v2.1.1
	github.com/gojuno/minimock/v3 v3.1.3
	github.com/google/uuid v1.3.0
	github.com/gorilla/mux v1.8.0
	github.com/iancoleman/strcase v0.2.0
	github.com/ilyakaznacheev/cleanenv v1.4.2
	github.com/joho/godotenv v1.5.1
	github.com/newrelic/go-agent/v3 v3.22.1
	github.com/newrelic/go-agent/v3/integrations/logcontext-v2/nrlogrus v1.0.0
	github.com/redis/go-redis/v9 v9.0.4
	github.com/sirupsen/logrus v1.9.2
	github.com/spf13/cobra v1.7.0
	github.com/spf13/viper v1.15.0
	github.com/stretchr/testify v1.8.4
	github.com/xitongsys/parquet-go v1.6.2
	github.com/xitongsys/parquet-go-source v0.0.0-20230607234618-40034c8066df
	gorm.io/driver/mysql v1.5.1
	gorm.io/driver/postgres v1.5.2
	gorm.io/gorm v1.25.1
)

require (
	github.com/BurntSushi/toml v1.2.1 // indirect
	github.com/apache/arrow/go/arrow v0.0.0-20200730104253-651201b0f516 // indirect
	github.com/apache/thrift v0.16.0 // indirect
	github.com/cespare/xxhash/v2 v2.2.0 // indirect
	github.com/davecgh/go-spew v1.1.1 // indirect
	github.com/dgryski/go-rendezvous v0.0.0-20200823014737-9f7001d12a5f // indirect
	github.com/fsnotify/fsnotify v1.6.0 // indirect
	github.com/go-sql-driver/mysql v1.7.1 // indirect
	github.com/golang/protobuf v1.5.3 // indirect
	github.com/golang/snappy v0.0.4 // indirect
	github.com/hashicorp/hcl v1.0.0 // indirect
	github.com/inconshreveable/mousetrap v1.1.0 // indirect
	github.com/jackc/pgpassfile v1.0.0 // indirect
	github.com/jackc/pgservicefile v0.0.0-20221227161230-091c0ba34f0a // indirect
	github.com/jackc/pgx/v5 v5.3.1 // indirect
	github.com/jinzhu/inflection v1.0.0 // indirect
	github.com/jinzhu/now v1.1.5 // indirect
	github.com/jmespath/go-jmespath v0.4.0 // indirect
	github.com/klauspost/compress v1.15.9 // indirect
	github.com/magiconair/properties v1.8.7 // indirect
	github.com/mitchellh/mapstructure v1.5.0 // indirect
	github.com/pelletier/go-toml/v2 v2.0.8 // indirect
	github.com/pierrec/lz4/v4 v4.1.15 // indirect
	github.com/pkg/errors v0.9.1 // indirect
	github.com/pmezard/go-difflib v1.0.0 // indirect
	github.com/spf13/afero v1.9.5 // indirect
	github.com/spf13/cast v1.5.1 // indirect
	github.com/spf13/jwalterweatherman v1.1.0 // indirect
	github.com/spf13/pflag v1.0.5 // indirect
	github.com/subosito/gotenv v1.4.2 // indirect
	golang.org/x/crypto v0.9.0 // indirect
	golang.org/x/net v0.10.0 // indirect
	golang.org/x/sys v0.9.0 // indirect
	golang.org/x/text v0.9.0 // indirect
	golang.org/x/xerrors v0.0.0-20220907171357-04be3eba64a2 // indirect
	google.golang.org/genproto v0.0.0-20230331144136-dcfb400f0633 // indirect
	google.golang.org/grpc v1.54.0 // indirect
	google.golang.org/protobuf v1.30.0 // indirect
	gopkg.in/ini.v1 v1.67.0 // indirect
	gopkg.in/square/go-jose.v2 v2.6.0 // indirect
	gopkg.in/yaml.v3 v3.0.1 // indirect
	olympos.io/encoding/edn v0.0.0-20201019073823-d3554ca0b0a3 // indirect
)

```

**Comment by marius-xentral** [comment]
ok, I figured it out. the auth0 host address that I use set up the validator HAS to end with a trailing frontslash. I recently set up those kind of parameters in vault and did not add the slash at the end (before that they were coming from my local env variables where I apparently did not forget them). So it is not a bug and only a result of bad configuration on my part. 

However, the error message that I received was very misleading. Maybe this could be adjusted. Also, I think the package could also take care of such a detail automatically.
